packages/@aws-cdk/cloud-assembly-schema/lib/assembly-manifest.ts

```diff
@@ -0,0 +1,340 @@
+interface BaseAssetMetadataEntry {
+    /**
+     * Requested packaging style
+     */
+    readonly packaging: string;
+
+    /**
+     * Logical identifier for the asset
+     */
+    readonly id: string;
+
+    /**
+     * The hash of the source directory used to build the asset.
+     */
+    readonly sourceHash: string;
+
+    /**
+     * Path on disk to the asset
+     */
+    readonly path: string;
+
+}
+
+/**
+ * Metadata Entry spec for files.
+ */
+export interface FileAssetMetadataEntry extends BaseAssetMetadataEntry {
+    /**
+     * Requested packaging style
+     */
+    readonly packaging: 'zip' | 'file';
+
+    /**
+     * Name of parameter where S3 bucket should be passed in
+     */
+    readonly s3BucketParameter: string;
+
+    /**
+     * Name of parameter where S3 key should be passed in
+     */
+    readonly s3KeyParameter: string;
+
+    /**
+     * The name of the parameter where the hash of the bundled asset should be passed in.
+     */
+    readonly artifactHashParameter: string;
+}
+
+/**
+ * Metadata Entry spec for container images.
+ */
+export interface ContainerImageAssetMetadataEntry extends BaseAssetMetadataEntry {
+    /**
+     * Type of asset
+     */
+    readonly packaging: 'container-image';
+
+    /**
+     * ECR Repository name and repo digest (separated by "@sha256:") where this
+     * image is stored.
+     *
+     * @default undefined If not specified, `repositoryName` and `imageTag` are
+     * required because otherwise how will the stack know where to find the asset,
+     * ha?
+     * @deprecated specify `repositoryName` and `imageTag` instead, and then you
+     * know where the image will go.
```

Comment:
any chance we can clean this up, or would it be a breaking change?

Reply:
You mean remove it altogether? It would be a breaking change. I'd rather not attempt to do these types of cleanups in this PR, too risky.

Reply:
Ok